### PR TITLE
[POST] /api/groups done

### PR DESCRIPTION
**Pull request by Wally5077**
#134

### CODE REVIEW

File: Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/GroupControllerIT.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @Autowired
+    private GroupDataRepository repository;
+
+    @AfterEach
+    void cleanUp() {
+        repository.deleteAll();
+    }
+
+    @Test
+    public void GiveOneUserCreateGroup_WhenGroupTitleUnique_ShouldCreateSuccessfully() throws Exception {
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/GroupControllerIT.java:39]: `WhenCreateGroupWithUniqueTitle_ShouldCreateSuccessfully`

File: Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/GroupControllerIT.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        ResultActions resultActions = createGroupRequest()
+                .andExpect(status().isOk());
+        GroupData groupData = getBody(resultActions, GroupData.class);
+        assertEquals(TEST_TITLE, groupData.getTitle());
+    }
+
+    @Test
+    public void GiveOneUserCreateGroup_WhenGroupTitleExist_ShouldCreateFailed() throws Exception {
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/GroupControllerIT.java:47]: `GiveOneGroupCreated_WhenCreateGroupWithDuplicateName_ShouldRejectWithBadRequest`

File: Spring-Boot/Spring-Boot-Student/src/main/java/tw/waterball/judgegirl/springboot/student/repositories/jpa/GroupData.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+public class GroupData {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Integer id;
+
+    private String title;
```
Comment (Johnny850807) [inline Spring-Boot/Spring-Boot-Student/src/main/java/tw/waterball/judgegirl/springboot/student/repositories/jpa/GroupData.java:25]: Sorry, my mistake, it should be `name`.

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done.

**Review by Chaoyuuu — APPROVED** [review]

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**